### ISSUE TITLE
Constant items given to players

### ISSUE THREAD

**Issue by LPGamersRob** [WooMinecraft/WooMinecraft#99]
i have got this working on my server and have setup a test item that uses /say and /give and is ment to give only 64 diamonds and say you bought a thingy once but it is doing it over and over. and it has filled my inv with diamonds in just under 2 hours. also in my console i keep getting 02.07 00:59:25 [Server] WARN [WooMinecraft] A JSONObject text must begin with '{' at 1 [character 2 line 1]

is this a bug.

**Comment by JayWood** [comment]
Definitely a bug, but not sure where or how. The last error `[WooMinecraft] A JSONObject text must begin with '{' at 1 [character 2 line 1]` sounds like your WordPress server isn't responding properly.

But, since you are indeed getting diamonds that tells me the error is happening during the second transaction.

WMC does two things, it first reads the JSON feed ( without that commands can't run ) it then sends data back to the server saying which order ID's were processed. It would appear the second step is breaking somehow.

If you can provide your server key and WordPress URL I could run some tests to see where it's breaking hopefully.

**Comment by LPGamersRob** [comment]
i have now also learned that if i do /woo check it also makes this happen and if a person has bought more than one item from the shop it gives them every items over and over. and i will pm you my server key over now

**Comment by JayWood** [comment]
@LPGamersRob if you have a spigot account you can contact me here: https://www.spigotmc.org/members/phyrax.133345/

**Comment by LPGamersRob** [comment]
i have sent you over my url and key so you can take a look ?

**Comment by JayWood** [comment]
@LPGamersRob as I suspected, I'm getting an error when POSTing data to the server. There's this nifty thing call POSTman which I can use to send data and test things. The only thing this does is say "hey this order's commands have been sent, remove it from the feed".

However, when I send that request to your server, it responds with an HTTP 503 error. Which usually means your server is freaking out, or WordPress has a problem.

Given the extensible nature of WordPress, it's possible your theme or a plugin is interfering with normal operations and there's no way for me to account for all things unfortunately. So during testing I only test using the default 2016 theme, and no plugins active.

See attached image:

![error-503](https://cloud.githubusercontent.com/assets/2303074/16537622/90fa31aa-3fd5-11e6-8915-3331b0b777ee.png)

Unfortunately given the results, i cannot troubleshoot further. It is possible that WordPress or your server has logged this 503 error somewhere.  Typically if you have debugging enabled it will be in `wp-content/debug.log` - if you do not, it's in your PHP error log, which varies server to server.

If you happen to narrow down the plugin that's causing the issue I can install it on my dev environment and test from there, or if it's your theme, I can do that as well, but since I don't have full root access to your server, I can't do much more. ( nor do I want root access to your server ).

**Comment by LPGamersRob** [comment]
Thank you for checking that i have now removed the theme that i was using on my website if that would be any help to you ?, and what i don't understand why the server is still giving out items every 2 mins as i have that set in my config, is there supposed to be some form of protection in the plugin to stop this ?

**Comment by LPGamersRob** [comment]
just so that you know i have set my wordpress to completely simple. there are no plugins on here at all apart from woocommerse and woocraft and am still getting the problem, Thanks

**Comment by JayWood** [comment]
The reason items are being given out more than once is due to the fact the Minecraft server cannot POST to your WordPress server to let WordPress know that the orders are delivered.

Without Java telling WordPress " hey I sent this order, remove it from the feed" - basically - you'll get the same commands every time.

What has to be done at this point your web developer or your web host should investigate the reason for the 503 on your server when posting data to the plugin. The first step would be to deactivate ALL plugins but WMC, and use the default 2016 theme.  Then, test using POSTman settings I provided above.

If you're still getting a 503 error, you'll then have to review the debug log on WordPress or your server. If you're not familiar how to debug in WordPress, checkout this [article I wrote for WebDevStudios](https://webdevstudios.com/2015/10/15/debugging-wordpress-tips-snippets/).

Bottom line, at this point it's a server issue which I cannot troubleshoot any further, I wish I could be of more help. :disappointed:
